Makefile

```diff
@@ -298,6 +298,25 @@ up-detach: build-services ## Bring up local Grapl and detach to return control t
 	docker-compose \
 		--file docker-compose.yml \
 		up --detach --force-recreate
+	# Wait for provisioning to fully complete before exiting. We previously
+	# ran wait-for-it in each of the test containers themselves, but this
+	# makes them depenedent on the local Grapl environment. Moving that
+	# here helps make the tests be able to run against a deployment in AWS.
+	# We're using grapl-e2e-tests from ./test/docker-compose.e2e-tests.yml,
+	# although we don't need to use this particular service. We just a
+	# service with an image that has wait-for-it installed, and it doesn't
+	# seem worthwhile to define a new service in a compose file just for
+	# that. This is probably a temporary solution anyway until we finalize
+	# the changes to local provisioning.
+	#
+	# It looks like docker-compose isn't honoring COMPOSE_IGNORE_ORPHANS
+	# for the 'run' command, so we're left with a bogus warning. This looks
+	# related: https://github.com/docker/compose/issues/8203.
+	docker-compose \
+		--file ./test/docker-compose.e2e-tests.yml \
+		run --rm \
+		grapl-e2e-tests \
+		wait-for-it grapl-engagement-view-uploader:$${WAIT_PORT} --timeout=250
```

Comment:
🤔 I don't know... I think having a container just for `wait-for-it` makes a lot of sense, in that it's very focused and the intent behind it and its usage is very clear. Nobody has to wonder why we're using the e2e-tests container for this, you don't have to explain it, and nothing mysteriously breaks if we remove `wait-for-it` from that container. It also doesn't create unnecessary dependencies between the main docker-compose file and the e2e one.

Such a container would be pretty simple, too. Here's one I found just now: https://github.com/whs/docker-wait-for-it/blob/master/Dockerfile (I'm not suggesting we directly use that one; just illustrating it's not an onerous task.)

Alternatively, we might be able to do something by listening for Docker events on the `grapl-engagement-view-uploader` container: https://docs.docker.com/engine/reference/commandline/events/

Reply:
Fair enough.

Instead of pulling from some randos DockerHub, I'd prefer to pull from official debian and use apt to install, as we do elsewhere.

I'll push a change for this.

Reply:
I agree, moving this to the make target makes it less transparent. In the long term we'll build healthchecks to accomplish the startup ordering task but for now having containers wait on their dependencies with `wait-for-it.sh` is straightforward.

Reply:
This should be resolved now.